--- -	Fri Jan 18 12:11:36 2002
+++ kbd.c	Fri Jan 18 12:07:58 2002
@@ -58,7 +58,11 @@
 {
 	int w = 0;
 	if (!len) return;
-	if (!itrm->eqlen && can_write(itrm->sock_out) && (w = write(itrm->sock_out, data, len)) <= 0) {
+	if (!itrm->eqlen
+	    && can_write(itrm->sock_out)
+	    && (w = write(itrm->sock_out, data, len)) <= 0
+	    && (len > PIPE_BUF || errno != EAGAIN || (w = 0) )
+	   ) {
 		/*free_trm(itrm);*/
 		register_bottom_half((void (*)(void *))free_trm, itrm);
 		return;
